packages/material-ui-unstyled/src/FormControlUnstyled/FormControlUnstyled.tsx

```diff
@@ -0,0 +1,257 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import { unstable_isMuiElement as isMuiElement } from '@material-ui/utils';
+import FormControlContext from './FormControlContext';
+import { FormControlUnstyledProps } from './FormControlUnstyledProps';
+import appendStyleProps from '../utils/appendStyleProps';
+import { FormControlState } from './FormControlState';
+import isFieldFilled from '../utils/isFieldFilled';
+import classes from './formControlUnstyledClasses';
+
+/**
+ * Provides context such as filled/focused/error/required for form inputs.
+ * Relying on the context provides high flexibility and ensures that the state always stays
+ * consistent across the children of the `FormControl`.
+ * This context is used by the following components:
+ *
+ * *   FormLabel
+ * *   FormHelperText
+ * *   Input
+ * *   InputLabel
+ *
+ * You can find one composition example below and more going to [the demos](https://material-ui.com/components/text-fields/#components).
+ *
+ * ```jsx
+ * <FormControl>
+ *   <InputLabel htmlFor="my-input">Email address</InputLabel>
+ *   <Input id="my-input" aria-describedby="my-helper-text" />
+ *   <FormHelperText id="my-helper-text">We'll never share your email.</FormHelperText>
+ * </FormControl>
+ * ```
+ *
+ * ⚠️ Only one `InputBase` can be used within a FormControl because it create visual inconsistencies.
+ * For instance, only one input can be focused at the same time, the state shouldn't be shared.
+ *
+ * Demos:
+ *
+ * - [Text Fields](https://material-ui.com/components/text-fields/)
+ *
+ * API:
+ *
+ * - [FormControlUnstyled API](https://material-ui.com/api/form-control-unstyled/)
+ */
+const FormControl = React.forwardRef(function FormControl(
+  props: FormControlUnstyledProps,
+  ref: React.ForwardedRef<any>,
+) {
+  const {
+    children,
+    className,
+    component,
+    components = {},
+    componentsProps = {},
+    disabled = false,
+    error = false,
+    extraContextProperties = {},
+    focused: visuallyFocused,
+    hiddenLabel = false,
+    required = false,
+    ...other
+  } = props;
+
+  const styleProps: FormControlUnstyledProps = {
+    ...props,
+    disabled,
+    error,
+    hiddenLabel,
+    required,
+  };
+
+  const [adornedStart, setAdornedStart] = React.useState(() => {
```

Comment:
This is also related to Material design. Looks unrelated to an "unstyled" version.

Reply:
Though I remember that we also needed to clarify what an "adornment" is and what a "suffix" is in the TextField. So there is potentially a place for a generic "adornment" (which is probably better named as "secondary action") and "suffix" (e.g. a currency symbol) in the unstyled package.